pallets/circuit-portal/src/lib.rs

```diff
@@ -263,6 +266,95 @@ pub mod pallet {
             Ok(res)
         }
 
+        // #[pallet::weight(<T as pallet::Config<I>>::WeightInfo::submit_finality_proof(
+        // block_hash.len() as u32,
+        // block_hash.len() as u32,
+        // ))]
+        #[pallet::weight(< T as Config >::WeightInfo::register_gateway_default_polka())]
+        pub fn submit_parachain_header(
```

Comment:
as this is the parachain equivalent to `submit_header_range` (meant for relaychain header ranges; maybe renaming required) probably should be colocated in one pallet with `submit_header_range` and  `submit_finality_proof`.